### ISSUE TITLE
Misleading error message when flag name has an underscore

### ISSUE THREAD

**Issue by RichardBowenGM** [BrianPugh/cyclopts#110]
Example code:

``` python
from cyclopts import App
app = App()

@app.command
def foo(test_flag: bool):
  print("x" if test_flag else "y")

app()
```

Executing with `--test-flag`:

```
$python test.py foo --test_flag
╭─ Error ─────────────────────────────────────────────────────────────────╮
│ Unknown option: "--test-flag".                                          │
╰─────────────────────────────────────────────────────────────────────────╯
```

The error message seems incorrect, because `--test-flag` is right and is a known option. Ideally `cyclopts` should say `Unknown option: --test_flag`, or maybe `test-flag` and `test_flag` should be synonyms.

Version info: `cyclopts 2.3.1`, `python 3.9.18`.

**Comment by BrianPugh** [comment]
I'll be trying to pipe through what the CLI-provided token was so that help/error messages can be a bit more specific.

**Comment by BrianPugh** [comment]
This has been addressed in the newly released v2.4.0
